care/apps/accounts/serializers.py

```diff
@@ -1,10 +1,75 @@
-from rest_framework.serializers import ModelSerializer
+from django.db.models.query_utils import Q
+from django.utils.translation import ugettext as _
+
+from rest_framework import serializers as rest_serializers
+from rest_framework.authtoken.models import Token
 
 from apps.accounts import models as accounts_models
 
 
-class UserSerializer(ModelSerializer):
+class UserSerializer(rest_serializers.ModelSerializer):
 
     class Meta:
         model = accounts_models.User
         fields = '__all__'
+
+
+class StateSerializer(rest_serializers.ModelSerializer):
+    """
+    Serializer for state model
+    """
+    class Meta:
+        model = accounts_models.State
+        fields = ('name',)
+
+
+class DistrictSerializer(rest_serializers.ModelSerializer):
+    """
+    Serializer for state model
+    """
+    class Meta:
+        model = accounts_models.District
+        fields = ('name',)
+
+
+class LoginSerializer(rest_serializers.Serializer):
+    """
+    User login serializer
+    """
+    username = rest_serializers.CharField(label=_('User Name'))
+    password = rest_serializers.CharField(label=_('Password'), style={'input_type': 'password'})
+
+    def validate(self, attrs):
+        username = attrs.get('username')
+        password = attrs.get('password')
+
+        if username and password:
+            user = accounts_models.User.objects.filter(
+                Q(username__exact=username) | Q(email__iexact=username)
+            ).first()
```

Comment:
Don't use .first()

Reply:
Should I use .get() inside try except here?